File: 2021/runner.rb
require "benchmark"

module AOC
    class AocDay
        attr_reader :p1, :p2

        FORMAT = "%n #{Benchmark::FORMAT}"

        def initialize(lines) 
            @lines = lines
        end

        def run()
            p1time = Benchmark.measure("Part 1") {part1}
            p2time = Benchmark.measure("Part 2") {part2}

            puts "Timings: #{Benchmark::CAPTION}"
            puts p1time.format(FORMAT)
            puts p2time.format(FORMAT)
        end

        def part1()
            @p1 = "TODO: Compute"
        end

        def part2() 
            @p2 = "TODO: Compute"
        end
    end
end

Dir["./days/*.rb"].each{|f| require f}

lines = File.readlines("inputs/d#{ARGV[0]}.txt").map(&:strip)
day = "AOC::Day#{ARGV[0]}".split('::').inject(Object) {|o,c| o.const_get c}.new lines

day.run

puts "Part 1: #{day.p1}"
puts "Part 2: #{day.p2}"

File: 2021/Python/runner.py
import sys
import days
from timeit import timeit
from utils import AoCDay

def run(num: str):
    day: AoCDay
    try:
        with open(f"input/d{num}.txt") as f:
            dayClass: AoCDay.__class__
            try:
                dayClass = getattr(days, f"Day{num}")
            except:
                return print(f"Day {num} not found")

            day = dayClass(f.read())            
    except FileNotFoundError:
        return print(f"Day {num} not found in days folder")

    day.part1()
    print(f"Part 1: {day.p1}")

    day.part2()
    print(f"Part 2: {day.p2}")

try:
    dayNumber = sys.argv[1]
except:
    print("Please write the number day you wish to run")

run(dayNumber)
    